To fix: JavaLayerConfigurationsHelperTest

<!-- Thanks for filing an issue! Please use this template to provide more context for the issue . -->

**Description of the issue**: Test failure

**Expected behavior**: Test success

**Steps to reproduce**: <!-- Please provide a minimal and precise series of steps -->

```shell
cd jib-maven-plugin
./mvnw verify -Pintegration-tests
```

**Environment**: <!-- OS, Maven, and other relevant environment information -->

Linux rodete

**Log output**: <!-- If applicable, provide relevant log output -->

```
[INFO] Running com.google.cloud.tools.jib.plugins.common.JavaLayerConfigurationsHelperTest
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.004 s <<< FAILURE! - in com.google.cloud.tools.jib.plugins.common.JavaLayerConfigurationsHelperTest
[ERROR] testFromExplodedWar(com.google.cloud.tools.jib.plugins.common.JavaLayerConfigurationsHelperTest)  Time elapsed: 0.004 s  <<< FAILURE!
java.lang.AssertionError: expected:<[/usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/META-INF, /usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/META-INF/context.xml, /usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/Test.jsp, /usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/WEB-INF, /usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/WEB-INF/classes, /usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/WEB-INF/classes/empty_dir, /usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/WEB-INF/classes/package, /usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/WEB-INF/classes/package/test.properties, /usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/WEB-INF/lib, /usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/WEB-INF/web.xml]> but was:<[/usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/META-INF, /usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/META-INF/context.xml, /usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/Test.jsp, /usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/WEB-INF, /usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/WEB-INF/classes, /usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/WEB-INF/classes/empty_dir, /usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/WEB-INF/classes/package, /usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/WEB-INF/classes/package/test.properties, /usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/WEB-INF/lib, /usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/WEB-INF/web.xml, /usr/local/google/home/qingyangc/clones/jib/jib-maven-plugin/target/test-classes/exploded-war/file]>
	at com.google.cloud.tools.jib.plugins.common.JavaLayerConfigurationsHelperTest.assertLayerEntriesUnordered(JavaLayerConfigurationsHelperTest.java:42)
	at com.google.cloud.tools.jib.plugins.common.JavaLayerConfigurationsHelperTest.assertSourcePathsUnordered(JavaLayerConfigurationsHelperTest.java:47)
	at com.google.cloud.tools.jib.plugins.common.JavaLayerConfigurationsHelperTest.testFromExplodedWar(JavaLayerConfigurationsHelperTest.java:72)
```

--- a/jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/JavaLayerConfigurationsHelperTest.java
+++ b/jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/JavaLayerConfigurationsHelperTest.java
@@ -25,16 +25,22 @@ import java.net.URISyntaxException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import java.util.function.Function;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 import org.junit.Assert;
+import org.junit.ClassRule;
 import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
 
 /** Tests for {@link JavaLayerConfigurationsHelper}. */
 public class JavaLayerConfigurationsHelperTest {
+
+  @ClassRule public static TemporaryFolder temporaryFolder = new TemporaryFolder();
 
   private static <T> void assertLayerEntriesUnordered(
       List<T> expectedPaths, List<LayerEntry> entries, Function<LayerEntry, T> fieldSelector) {
@@ -56,43 +62,59 @@ public class JavaLayerConfigurationsHelperTest {
 
   @Test
   public void testFromExplodedWar() throws URISyntaxException, IOException {
-    Path explodedWar = Paths.get(Resources.getResource("exploded-war").toURI());
-    Files.createDirectories(explodedWar.resolve("WEB-INF/classes/empty_dir"));
+    // Copy test files to a temporary directory that we can safely operate on
+    Path temporaryExplodedWar = temporaryFolder.newFolder("exploded-war").toPath();
+    Path resourceExplodedWar = Paths.get(Resources.getResource("exploded-war").toURI());
+    try (Stream<Path> stream = Files.walk(resourceExplodedWar)) {
+      stream.forEach(
+          source -> {
+            try {
+              Files.copy(
+                  source,
+                  temporaryExplodedWar.resolve(resourceExplodedWar.relativize(source)),
+                  StandardCopyOption.REPLACE_EXISTING);
+            } catch (IOException ex) {
+              Assert.fail("Failed to copy resources to temp directory: " + ex.getMessage());
+            }
+          });
+    }
+
+    Files.createDirectories(temporaryExplodedWar.resolve("WEB-INF/classes/empty_dir"));
     Path extraFilesDirectory = Paths.get(Resources.getResource("layer").toURI());
 
     JavaLayerConfigurations configuration =
         JavaLayerConfigurationsHelper.fromExplodedWar(
-            explodedWar,
+            temporaryExplodedWar,
             AbsoluteUnixPath.get("/my/app"),
             extraFilesDirectory,
             Collections.emptyMap());
 
     assertSourcePathsUnordered(
-        Collections.singletonList(explodedWar.resolve("WEB-INF/lib/dependency-1.0.0.jar")),
+        Collections.singletonList(temporaryExplodedWar.resolve("WEB-INF/lib/dependency-1.0.0.jar")),
         configuration.getDependencyLayerEntries());
     assertSourcePathsUnordered(
         Collections.singletonList(
-            explodedWar.resolve("WEB-INF/lib/dependencyX-1.0.0-SNAPSHOT.jar")),
+            temporaryExplodedWar.resolve("WEB-INF/lib/dependencyX-1.0.0-SNAPSHOT.jar")),
         configuration.getSnapshotDependencyLayerEntries());
     assertSourcePathsUnordered(
         Arrays.asList(
-            explodedWar.resolve("META-INF"),
-            explodedWar.resolve("META-INF/context.xml"),
-            explodedWar.resolve("Test.jsp"),
-            explodedWar.resolve("WEB-INF"),
-            explodedWar.resolve("WEB-INF/classes"),
-            explodedWar.resolve("WEB-INF/classes/empty_dir"),
-            explodedWar.resolve("WEB-INF/classes/package"),
-            explodedWar.resolve("WEB-INF/classes/package/test.properties"),
-            explodedWar.resolve("WEB-INF/lib"),
-            explodedWar.resolve("WEB-INF/web.xml")),
+            temporaryExplodedWar.resolve("META-INF"),
+            temporaryExplodedWar.resolve("META-INF/context.xml"),
+            temporaryExplodedWar.resolve("Test.jsp"),
+            temporaryExplodedWar.resolve("WEB-INF"),
+            temporaryExplodedWar.resolve("WEB-INF/classes"),
+            temporaryExplodedWar.resolve("WEB-INF/classes/empty_dir"),
+            temporaryExplodedWar.resolve("WEB-INF/classes/package"),
+            temporaryExplodedWar.resolve("WEB-INF/classes/package/test.properties"),
+            temporaryExplodedWar.resolve("WEB-INF/lib"),
+            temporaryExplodedWar.resolve("WEB-INF/web.xml")),
         configuration.getResourceLayerEntries());
     assertSourcePathsUnordered(
         Arrays.asList(
-            explodedWar.resolve("WEB-INF/classes/HelloWorld.class"),
-            explodedWar.resolve("WEB-INF/classes/empty_dir"),
-            explodedWar.resolve("WEB-INF/classes/package"),
-            explodedWar.resolve("WEB-INF/classes/package/Other.class")),
+            temporaryExplodedWar.resolve("WEB-INF/classes/HelloWorld.class"),
+            temporaryExplodedWar.resolve("WEB-INF/classes/empty_dir"),
+            temporaryExplodedWar.resolve("WEB-INF/classes/package"),
+            temporaryExplodedWar.resolve("WEB-INF/classes/package/Other.class")),
         configuration.getClassLayerEntries());
     assertSourcePathsUnordered(
         Arrays.asList(

--- a/jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/JavaLayerConfigurationsHelperTest.java
+++ b/jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/JavaLayerConfigurationsHelperTest.java
@@ -17,30 +17,28 @@
 package com.google.cloud.tools.jib.plugins.common;
 
 import com.google.cloud.tools.jib.filesystem.AbsoluteUnixPath;
+import com.google.cloud.tools.jib.filesystem.FileOperations;
 import com.google.cloud.tools.jib.frontend.JavaLayerConfigurations;
 import com.google.cloud.tools.jib.image.LayerEntry;
+import com.google.common.collect.ImmutableList;
 import com.google.common.io.Resources;
 import java.io.IOException;
 import java.net.URISyntaxException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.nio.file.StandardCopyOption;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import java.util.function.Function;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
 import org.junit.Assert;
-import org.junit.ClassRule;
+import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
 
 /** Tests for {@link JavaLayerConfigurationsHelper}. */
 public class JavaLayerConfigurationsHelperTest {
-
-  @ClassRule public static TemporaryFolder temporaryFolder = new TemporaryFolder();
 
   private static <T> void assertLayerEntriesUnordered(
       List<T> expectedPaths, List<LayerEntry> entries, Function<LayerEntry, T> fieldSelector) {
@@ -60,24 +58,14 @@ public class JavaLayerConfigurationsHelperTest {
         expectedPaths, entries, layerEntry -> layerEntry.getExtractionPath().toString());
   }
 
+  @Rule public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
   @Test
   public void testFromExplodedWar() throws URISyntaxException, IOException {
     // Copy test files to a temporary directory that we can safely operate on
-    Path temporaryExplodedWar = temporaryFolder.newFolder("exploded-war").toPath();
     Path resourceExplodedWar = Paths.get(Resources.getResource("exploded-war").toURI());
-    try (Stream<Path> stream = Files.walk(resourceExplodedWar)) {
-      stream.forEach(
-          source -> {
-            try {
-              Files.copy(
-                  source,
-                  temporaryExplodedWar.resolve(resourceExplodedWar.relativize(source)),
-                  StandardCopyOption.REPLACE_EXISTING);
-            } catch (IOException ex) {
-              Assert.fail("Failed to copy resources to temp directory: " + ex.getMessage());
-            }
-          });
-    }
+    FileOperations.copy(ImmutableList.of(resourceExplodedWar), temporaryFolder.getRoot().toPath());
+    Path temporaryExplodedWar = temporaryFolder.getRoot().toPath().resolve("exploded-war");
 
     Files.createDirectories(temporaryExplodedWar.resolve("WEB-INF/classes/empty_dir"));
     Path extraFilesDirectory = Paths.get(Resources.getResource("layer").toURI());

